Xamarin.Forms.Platform.iOS/Renderers/ImageButtonRenderer.cs

```diff
@@ -0,0 +1,146 @@
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Foundation;
+using UIKit;
+using Xamarin.Forms.PlatformConfiguration.iOSSpecific;
+using SizeF = CoreGraphics.CGSize;
+
+namespace Xamarin.Forms.Platform.iOS
+{
+	public class ImageButtonRenderer : ViewRenderer<ImageButton, UIButton>, IImageVisualElementRenderer
+	{
+		bool _isDisposed;
+
+		// This looks like it should be a const under iOS Classic,
+		// but that doesn't work under iOS 
+		// ReSharper disable once BuiltInTypeReferenceStyle
+		// Under iOS Classic Resharper wants to suggest this use the built-in type ref
+		// but under iOS that suggestion won't work
+		readonly nfloat _minimumButtonHeight = 44; // Apple docs
+
+
+		public ImageButtonRenderer() : base()
+		{
+			ButtonElementManager.Init(this);
+			BorderElementManager.Init(this);
+			ImageElementManager.Init(this);
+		}
+
+		public override SizeF SizeThatFits(SizeF size)
+		{
+			var result = base.SizeThatFits(size);
+
+			if (result.Height < _minimumButtonHeight)
+			{
+				result.Height = _minimumButtonHeight;
```

Comment:
what if  want a really small button?

Reply:
This is what we use for the minimum desired height on iOS buttons as well
https://github.com/xamarin/Xamarin.Forms/blob/bd31e1e9fc8b2f9ad94cc99e0c7ab058174821f3/Xamarin.Forms.Platform.iOS/Renderers/ButtonRenderer.cs#L27

The constraints still get set to really low numbers if you specify them though and that's how they get measured out

Example
![image](https://user-images.githubusercontent.com/5375137/46894389-8d0da080-ce31-11e8-9b5e-ba6dafaeff5e.png)